### PR TITLE
Bug 2516 fix v0

### PR DESCRIPTION
**Pull request by caitlin-hennessy**
Allows blocks to be built from Javascript syntax like `music.ringTone(music.noteFrequency(Note.D));` 

See https://github.com/Microsoft/pxt/issues/2516

### CODE REVIEW

File: pxtblocks/fields/field_note.ts [hunk 1]
```diff
@@ -185,7 +235,10 @@ namespace pxtblockly {
          * @param {string} note The new note in string format.
          */
         setValue(note: string) {
-            note = String(parseFloat(note || "0"));
+            let regex: RegExp = new RegExp("Note.(.*)");
```
Comment (riknoll) [inline pxtblocks/fields/field_note.ts]: Move this regex declaration outside of the `setValue`

File: pxtblocks/fields/field_note.ts [hunk 2]
```diff
@@ -185,7 +235,10 @@ namespace pxtblockly {
...
+            let match = regex.exec(note);
+            let noteName: any = (match && match.length > 1) ? match[1] : null;
```
Comment (riknoll) [inline pxtblocks/fields/field_note.ts]: We typically avoid using the `any` type anywhere in the code. I think you can just declare `string` here.

File: pxtblocks/fields/field_note.ts [hunk 3]
```diff
@@ -185,7 +235,10 @@ namespace pxtblockly {
...
+            note = Note[noteName] ? String(parseFloat(Note[noteName])) : String(parseFloat(note || "0"));
```
Comment (riknoll) [inline pxtblocks/fields/field_note.ts]: The first `parseFloat` call here isn't needed. I think it should already be a number

File: pxtblocks/fields/field_note.ts [hunk 4]
```diff
@@ -1,6 +1,56 @@
 /// <reference path="../../localtypings/blockly.d.ts" />
 
 namespace pxtblockly {
+        enum Note {
```
Comment (riknoll) [inline pxtblocks/fields/field_note.ts]: The tabbing looks to be a little off here

Comment (caitlin-hennessy) [inline pxtblocks/fields/field_note.ts]: Fixed

File: pxtblocks/fields/field_note.ts (hunk 3 above)
Comment (caitlin-hennessy) [inline pxtblocks/fields/field_note.ts]: Fixed

File: pxtblocks/fields/field_note.ts (hunk 1 above)
Comment (riknoll) [inline pxtblocks/fields/field_note.ts]: Oh, also, try `$Note\.(.+)^` instead

Comment (caitlin-hennessy) [inline pxtblocks/fields/field_note.ts]: Moved regex but left "any" as per our discussion.

**Pull request merged**